applications/StructuralMechanicsApplication/custom_constitutive/plane_stress_d_plus_d_minus_damage_masonry_2d.cpp

```diff
@@ -0,0 +1,734 @@
+	array_1d<double, 3>& rPredictiveStressVector, 
+	double& rEquivalentStress,
+	ConstitutiveLaw::Parameters& rValues
+	)
+{
+	const Properties& r_material_properties = rValues.GetMaterialProperties();
+	
+	const double yield_tension = r_material_properties[YIELD_STRESS_TENSION];
+	const double yield_compression = r_material_properties[YIELD_STRESS_COMPRESSION];
+	const double biaxial_compression_multiplier = r_material_properties[BIAXIAL_COMPRESSION_MULTIPLIER];
+	const double alpha = (biaxial_compression_multiplier - 1.0)/(2 * biaxial_compression_multiplier - 1.0);
+	const double alpha_factor = 1.0 / (1.0 - alpha);
+	const double beta = (yield_compression / yield_tension) * (1.0 - alpha) - (1.0 + alpha);
+	
+	double I1,J2;
+	ConstitutiveLawUtilities<3>::CalculateI1Invariant(rPredictiveStressVector, I1);
+	array_1d<double, 3> deviator = ZeroVector(3);
+	ConstitutiveLawUtilities<3>::CalculateJ2Invariant(rPredictiveStressVector, I1, deviator, J2);
+	
+	array_1d<double, 2> rPrincipalStressVector;
+	ConstitutiveLawUtilities<3>::CalculatePrincipalStresses(rPrincipalStressVector, rPredictiveStressVector);
+	const double principal_stress_1 = rPrincipalStressVector[0];
+	const double principal_stress_2 = rPrincipalStressVector[1];
+	
+	if (principal_stress_1 > 0.0){
+		rEquivalentStress = alpha_factor * (alpha*I1 + std::sqrt(3.0 * J2) + beta * principal_stress_1) * (yield_tension / yield_compression);
+	}	
+}
+/***********************************************************************************/
+/***********************************************************************************/
+void DamageDPlusDMinusMasonry2DLaw::CalculateEquivalentStressCompression(
+	array_1d<double, 3>& rPredictiveStressVector, 
+	double& rEquivalentStress,
+	ConstitutiveLaw::Parameters& rValues
+	)
+{
+	const Properties& r_material_properties = rValues.GetMaterialProperties();
+	
+	const double yield_tension = r_material_properties[YIELD_STRESS_TENSION];
+	const double yield_compression = r_material_properties[YIELD_STRESS_COMPRESSION];
+	const double biaxial_compression_multiplier = r_material_properties[BIAXIAL_COMPRESSION_MULTIPLIER];
+	const double shear_compression_reductor = r_material_properties[SHEAR_COMPRESSION_REDUCTOR];
+	
+    KRATOS_ERROR_IF(shear_compression_reductor < 0.0)<< "The SHEAR_COMPRESSION_REDUCTOR is supposed to be a value between 0.0 and 1.0" << std::endl;
+	KRATOS_ERROR_IF(shear_compression_reductor > 1.0)<< "The SHEAR_COMPRESSION_REDUCTOR is supposed to be a value between 0.0 and 1.0" << std::endl;
+	
+	const double alpha = (biaxial_compression_multiplier - 1.0)/(2.0* biaxial_compression_multiplier - 1.0);
+	const double alpha_factor = 1.0 / (1.0 - alpha);
+	const double beta = (yield_compression / yield_tension) * (1.0 - alpha) - (1.0 + alpha);
+	
+	double I1,I2,J2;
+	ConstitutiveLawUtilities<3>::CalculateI1Invariant(rPredictiveStressVector, I1);
+	array_1d<double, 3> deviator = ZeroVector(VoigtSize);

```

Comment:
If your VoigtSize is different than 3 this will be troublesome

Reply:
As far as I know, @philipkalkbrenner , this law is strictly in 2D right?

Reply:
yes, this one is strictly in 2D